src/api/informer.rs

```diff
@@ -110,14 +108,18 @@ where
     pub async fn init(self) -> Result<Self> {
         let initial = self.get_resource_version().await?;
         info!("Starting Informer for {:?}", self.resource);
-        *self.version.write().unwrap() = initial;
+        *self.version.lock().await = initial;
         Ok(self)
     }
 
     /// Initialize from a prior version
     pub fn init_from(self, v: String) -> Self {
         info!("Recreating Informer for {:?} at {}", self.resource, v);
-        *self.version.write().unwrap() = v;
+
+        // We need to block on this as our mutex needs go be async compatible
+        futures::executor::block_on(async {
+            *self.version.lock().await = v;
+        });
```

Comment:
This is probably a good use of `block_on` :+1: .

At init time, this won't block at all, and arguably the behaviour should be encapsulated in a constructor type pattern, but not found a good way of doing this neatly.